[FIX] l10n_ar_website_sale: Invoice date

zaoral opened the pull request:
### Description of the issue/feature this PR addresses:

If someone creates an invoice after 9 pm Argentina time, if the invoice is created automatically, then it is created with the date of the next day (UTC date) instead of today. This causes problems validating invoices in ARCA (ex AFIP), since when generating the invoice with the next day's date, no more invoices could be generated with today's date.

### Current behavior before PR:

1. Make e-commerce sales after 9 PM local time AR in a website set as automatic invoicing = True (journal by default it is an electronic journal)
2. Try to generate another invoice after that day, A simple way to test it is to make an invoice and then validate it.
3. We will receive ARCA (ex-AFIP) error telling us that we can not generate the invoice with today's date, we can only validate invoices with a date equal to or greater than tomorrow. 

### Desired behavior after PR is merged:

1. Make e-commerce sales after 9 PM local time AR in a website set as automatic invoicing = True (journal by default it is an electronic journal)
2. Try to generate another invoice after that day, A simple way to test it is to make an invoice and then validate it.
3. Manual invoice will be generated and validated in AFIP with today's date as the invoice date.


NOTE: We took the same approach that was used in the [POS module](https://github.com/odoo/odoo/blob/16.0/addons/point_of_sale/models/pos_order.py#L615-L631) to set the date, in this case, always forcing the Argentina timezone

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

robodoo commented:
[![Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/192808.png)](https://mergebot.odoo.com/odoo/odoo/pull/192808)

vbe-odoo commented:
@zaoral thanks for the PR.
I've tested and it LGTM.
Can you please add a a PR description (you could use the same as in the commit description).

@clbr-odoo this is ready for your technical review.

Thanks!

zaoral commented:
HI @vbe-odoo 
Thank you for the review

I have updated the PR description with all the details
Please let me know if it is ok or want to add something more.

Also, I have changed the PR from draft to normal PR to be checked by the technical team

NOTE: I have an extra related PR to this one https://github.com/odoo/enterprise/pull/77920. Initially, the PR was making changes but then I found out that it was not necessary so I closed it. Just let you know if you find weird runbot o robodoo on current PR 

Best Regards

clbr-odoo reviewed (approved):
Good job thank you :+1: 
@robodoo r+

robodoo commented:
@zaoral @clbr-odoo staging failed: ci/l10n on 4c347754630efd58a9460513e67da27233d395a1 (view more at https://runbot.odoo.com/runbot/build/74387419)

clbr-odoo commented:
@robodoo r-

vbe-odoo commented:
@clbr-odoo is there anything to fix on this PR? I see that runbot failed in the l10n test but not sure if the error is related to this PR

clbr-odoo commented:
Most likely the module was never tested before and the test now reveal a problem that was there, so it's not really linked to this but we can't merge it anyway.
Looking at the logs of the failing staging, we have a bit more information if you have time to check before me 
```odoo.addons.base.models.ir_qweb.QWebException: Error when compiling xml template
SyntaxError: t-elif directive must be preceded by t-if or t-elif directive
Template: l10n_ar.report_invoice_document
Path: /t/t/div[2]/div/table/thead/tr/th[9]/span[2]
Node: <span t-else=""/>
```

clbr-odoo commented:
https://github.com/odoo/odoo/pull/196765
You'll need to rebase on this commit when it's merged :slightly_smiling_face:

zaoral commented:
Hi @clbr-odoo 
Thanks for helping with the error!
I have just rebased and push on this PR,
now includes your change on https://github.com/odoo/odoo/pull/196765

Best Regrds

clbr-odoo commented:
Thank you !
@robodoo r+
